apis/druid/v1alpha1/druid_types.go

```diff
@@ -132,6 +148,9 @@ type DruidSpec struct {
 	// that is, it must match regex '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'
 	Nodes map[string]DruidNodeSpec `json:"nodes"`
 
+	// Operator deploys the sidecar container based on these properties. Sidecar will be deployed for all the Druid pods.
+	AdditionalContainer []AdditionalContainer `json:"additionalContainer,omitempty"`
+
```

Comment:
this is scoped at cluster scope only. so its right to assume that additional container will be common to all the nodes. IMHO its fine.